### PR TITLE
AP_Soaring: Coordinates system change

### PR DESCRIPTION
**Pull request by samuelctabor**
This simplifies the methods of tracking thermals. Previously the thermal position was estimated and logged as an offset from the aircraft location. This changes it to a north/east position in the local coordinate system.

This is based on my other PR https://github.com/ArduPilot/ardupilot/pull/11414

### CODE REVIEW

File: libraries/AP_Soaring/Variometer.cpp [hunk 1]
```diff
@@ -16,37 +16,50 @@ void Variometer::update(const float polar_K, const float polar_B, const float po
...
+    float sinkrate = correct_netto_rate(0.0f,  roll, _aspd_filt, polar_K, polar_Cd0, polar_B);
+
+    reading = dh + dsp*_aspd_filt/GRAVITY_MSS + sinkrate;
+    
+
+    filtered_reading = TE_FILT * reading + (1 - TE_FILT) * filtered_reading;                       // Apply low pass timeconst filter for noise
+    displayed_reading = TE_FILT_DISPLAYED * reading + (1 - TE_FILT_DISPLAYED) * displayed_reading;
+
+    _prev_update_time = AP_HAL::micros64();
+    new_data = true;
```
Comment (akolobov) [inline libraries/AP_Soaring/Variometer.cpp]: Nice work! A small question: why do you need the new_data variable in this class?

Comment (akolobov) [reply libraries/AP_Soaring/Variometer.cpp]: Nevermind, I see you remove it in another part of this change.

**Comment by magicrub** [comment]
If GPS is available, why not make it an absolute point?

**Comment by samuelctabor** [comment]
@akolobov there's also more changes in [this branch](https://github.com/samuelctabor/ardupilot/tree/SoaringUpgrades) and this one has been flight tested.

@magicrub could do, but the filter involves floating point operations and is better suited to NED coordinates, so I wanted to log the actual raw filter states.

**Comment by akolobov** [comment]
Yes, thanks @samuelctabor, I've carried some of them over to Frigatebird 2 days ago! Frigatebird currently doesn't use the Variometer class though, so I still need to hook it back in.

**Pull request closed without merging**